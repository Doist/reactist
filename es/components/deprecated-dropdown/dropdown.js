import {
    objectWithoutProperties as _objectWithoutProperties,
    objectSpread2 as _objectSpread2,
} from '../../_virtual/_rollupPluginBabelHelpers.js'
import * as React from 'react'
import ReactDOM from 'react-dom'
import classNames from 'classnames'
import { Button } from '../deprecated-button/deprecated-button.js'

const _excluded = ['children', 'onClick', 'tooltip', 'className']

class Box extends React.Component {
    constructor(props, context) {
        super(props, context)
        this._timeout = void 0

        this._handleClickOutside = (event) => {
            const dropdownDOMNode = ReactDOM.findDOMNode(this)
            if (dropdownDOMNode && !dropdownDOMNode.contains(event.target)) this._toggleShowBody()
            else if (!this.props.allowBodyInteractions) {
                // won't close when body interactions are allowed
                this._timeout = setTimeout(() => {
                    if (this.state.showBody) {
                        this._toggleShowBody()
                    }
                }, 100)
            }
        }

        this._toggleShowBody = () => {
            if (!this.state.showBody) {
                // will show
                if (this.props.onShowBody) this.props.onShowBody()
                document.addEventListener('click', this._handleClickOutside, true)
            } else {
                // will hide
                if (this.props.onHideBody) this.props.onHideBody()
                document.removeEventListener('click', this._handleClickOutside, true)
            }

            this.setState({
                showBody: !this.state.showBody,
            })
        }

        this._setPosition = (body) => {
            if (body) {
                const scrollingParent = document.getElementById(
                    this.props.scrolling_parent ? this.props.scrolling_parent : '',
                )

                if (scrollingParent) {
                    const dropdown = ReactDOM.findDOMNode(this)

                    if (!dropdown) {
                        return
                    }

                    const dropdownVerticalPosition = ReactDOM.findDOMNode(this).offsetTop
                    const dropdownTrigger = dropdown.querySelector('.trigger')

                    if (!dropdownTrigger) {
                        return
                    }

                    const dropdownTriggerHeight = dropdownTrigger.clientHeight
                    const dropdownBodyHeight = body.clientHeight
                    const scrollingParentHeight = scrollingParent.clientHeight
                    const scrollingParentOffset = scrollingParent.scrollTop
                    const bottomOffset =
                        scrollingParentHeight +
                        scrollingParentOffset -
                        dropdownVerticalPosition -
                        dropdownTriggerHeight
                    const top = bottomOffset < dropdownBodyHeight

                    if (top !== this.state.top) {
                        this.setState({
                            top,
                        })
                    }
                }
            }
        }

        this.state = {
            showBody: false,
            top: props.top || false,
        }
        this._timeout = undefined
    }

    componentWillUnmount() {
        document.removeEventListener('click', this._handleClickOutside, true)

        if (this._timeout) {
            clearTimeout(this._timeout)
        }
    }

    _getTriggerComponent() {
        var _this$props$children

        const _trigger =
            (_this$props$children = this.props.children) == null ? void 0 : _this$props$children[0]

        return _trigger
            ? /*#__PURE__*/ React.cloneElement(_trigger, {
                  onClick: this._toggleShowBody,
              })
            : undefined
    } // https://facebook.github.io/react/docs/refs-and-the-dom.html#exposing-dom-refs-to-parent-components

    _getBodyComponent() {
        if (!this.state.showBody) {
            return null
        }

        const { top } = this.state
        const { right = false, children } = this.props
        const props = {
            top,
            right,
            setPosition: this._setPosition,
        }
        const className = classNames({
            body_wrapper: true,
            with_arrow: true,
            top: top,
            bottom: !top,
        })
        const body = children == null ? void 0 : children[1]
        const contentMarkup =
            typeof body === 'function'
                ? body(props)
                : body
                ? /*#__PURE__*/ React.cloneElement(body, props)
                : undefined
        return /*#__PURE__*/ React.createElement(
            'div',
            {
                className: className,
                style: {
                    position: 'relative',
                },
            },
            contentMarkup,
        )
    }

    render() {
        const className = classNames('reactist_dropdown', this.props.className)
        const { top } = this.state
        return /*#__PURE__*/ React.createElement(
            'div',
            {
                style: {
                    display: 'inline-block',
                },
                className: className,
                'data-testid': 'reactist-dropdown-box',
            },
            top && this._getBodyComponent(),
            this._getTriggerComponent(),
            !top && this._getBodyComponent(),
        )
    }
}

Box.displayName = void 0
Box.displayName = 'Dropdown.Box'
const Trigger = /*#__PURE__*/ React.forwardRef(function Trigger(_ref, ref) {
    let { children, onClick, tooltip, className } = _ref,
        props = _objectWithoutProperties(_ref, _excluded)

    function handleClick(event) {
        event.preventDefault()
        event.stopPropagation()
        if (onClick) onClick(event)
    }

    return /*#__PURE__*/ React.createElement(
        Button,
        _objectSpread2(
            _objectSpread2({}, props),
            {},
            {
                className: classNames('trigger', className),
                onClick: handleClick,
                tooltip: tooltip,
                ref: ref,
            },
        ),
        children,
    )
})
Trigger.displayName = 'Dropdown.Trigger'

function Body({ top, right, children, setPosition }) {
    const style = {
        position: 'absolute',
        right: 0,
        top: 0,
    }

    if (top) {
        style.top = 'auto'
        style.bottom = 0
    }

    if (right) {
        style.right = 'auto'
        style.left = 0
    }

    return /*#__PURE__*/ React.createElement(
        'div',
        {
            ref: setPosition,
            style: style,
            className: 'body',
            id: 'reactist-dropdown-body',
            'data-testid': 'reactist-dropdown-body',
        },
        children,
    )
}

Body.displayName = 'Dropdown.Body'
const Dropdown = {
    Box,
    Trigger,
    Body,
}

export { Dropdown }
//# sourceMappingURL=dropdown.js.map
