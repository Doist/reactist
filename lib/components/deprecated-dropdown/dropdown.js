"use strict";Object.defineProperty(exports,"__esModule",{value:!0});var t=require("../../_virtual/_rollupPluginBabelHelpers.js"),e=require("react"),o=require("react-dom"),i=require("classnames"),s=require("../deprecated-button/deprecated-button.js");function r(t){return t&&"object"==typeof t&&"default"in t?t:{default:t}}function n(t){if(t&&t.__esModule)return t;var e=Object.create(null);return t&&Object.keys(t).forEach((function(o){if("default"!==o){var i=Object.getOwnPropertyDescriptor(t,o);Object.defineProperty(e,o,i.get?i:{enumerable:!0,get:function(){return t[o]}})}})),e.default=t,e}var d=n(e),l=r(o),a=r(i);const c=["children","onClick","tooltip","className"];class p extends d.Component{constructor(t,e){super(t,e),this._timeout=void 0,this._handleClickOutside=t=>{const e=l.default.findDOMNode(this);e&&!e.contains(t.target)?this._toggleShowBody():this.props.allowBodyInteractions||(this._timeout=setTimeout(()=>{this.state.showBody&&this._toggleShowBody()},100))},this._toggleShowBody=()=>{this.state.showBody?(this.props.onHideBody&&this.props.onHideBody(),document.removeEventListener("click",this._handleClickOutside,!0)):(this.props.onShowBody&&this.props.onShowBody(),document.addEventListener("click",this._handleClickOutside,!0)),this.setState({showBody:!this.state.showBody})},this._setPosition=t=>{if(t){const e=document.getElementById(this.props.scrolling_parent?this.props.scrolling_parent:"");if(e){const o=l.default.findDOMNode(this);if(!o)return;const i=l.default.findDOMNode(this).offsetTop,s=o.querySelector(".trigger");if(!s)return;const r=e.clientHeight+e.scrollTop-i-s.clientHeight<t.clientHeight;r!==this.state.top&&this.setState({top:r})}}},this.state={showBody:!1,top:t.top||!1},this._timeout=void 0}componentWillUnmount(){document.removeEventListener("click",this._handleClickOutside,!0),this._timeout&&clearTimeout(this._timeout)}_getTriggerComponent(){var t;const e=null==(t=this.props.children)?void 0:t[0];return e?d.cloneElement(e,{onClick:this._toggleShowBody}):void 0}_getBodyComponent(){if(!this.state.showBody)return null;const{top:t}=this.state,{right:e=!1,children:o}=this.props,i={top:t,right:e,setPosition:this._setPosition},s=a.default({body_wrapper:!0,with_arrow:!0,top:t,bottom:!t}),r=null==o?void 0:o[1],n="function"==typeof r?r(i):r?d.cloneElement(r,i):void 0;return d.createElement("div",{className:s,style:{position:"relative"}},n)}render(){const t=a.default("reactist_dropdown",this.props.className),{top:e}=this.state;return d.createElement("div",{style:{display:"inline-block"},className:t,"data-testid":"reactist-dropdown-box"},e&&this._getBodyComponent(),this._getTriggerComponent(),!e&&this._getBodyComponent())}}p.displayName=void 0,p.displayName="Dropdown.Box";const u=d.forwardRef((function(e,o){let{children:i,onClick:r,tooltip:n,className:l}=e,p=t.objectWithoutProperties(e,c);return d.createElement(s.Button,t.objectSpread2(t.objectSpread2({},p),{},{className:a.default("trigger",l),onClick:function(t){t.preventDefault(),t.stopPropagation(),r&&r(t)},tooltip:n,ref:o}),i)}));function h({top:t,right:e,children:o,setPosition:i}){const s={position:"absolute",right:0,top:0};return t&&(s.top="auto",s.bottom=0),e&&(s.right="auto",s.left=0),d.createElement("div",{ref:i,style:s,className:"body",id:"reactist-dropdown-body","data-testid":"reactist-dropdown-body"},o)}u.displayName="Dropdown.Trigger",h.displayName="Dropdown.Body",exports.Dropdown={Box:p,Trigger:u,Body:h};
//# sourceMappingURL=dropdown.js.map
