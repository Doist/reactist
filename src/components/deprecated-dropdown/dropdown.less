@import '../styles/constants.less';

.reactist_dropdown {
    .trigger {
        cursor: pointer;
        display: block;
    }

    .body {
        border-radius: 3px;
        border: 1px solid @color_border_grey;
        overflow: hidden;
        box-shadow: 0 1px 8px 0 rgba(0, 0, 0, 0.08);
        z-index: 1;
        background-color: #fff;
    }

    hr {
        border: none;
        height: 1px;
        background-color: @color_border_grey;
        margin: 0 5px;
    }

    .with_arrow:after,
    .with_arrow:before {
        z-index: 1;
        content: '';
        display: block;
        position: absolute;
        width: 0;
        height: 0;
        border-style: solid;
        border-width: 6px;
        right: 14px;
    }

    .with_arrow:after {
        top: -11px;
        border-color: transparent transparent #fff transparent;
    }

    .with_arrow:before {
        top: -12px;
        border-color: transparent transparent @color_border_grey transparent;
    }

    .with_arrow.top:after {
        top: -1px;
        border-color: #fff transparent transparent transparent;
    }

    .with_arrow.top:before {
        top: -1px;
        right: 13px;
        border-width: 7px;
        border-color: @color_border_grey transparent transparent transparent;
    }
}
