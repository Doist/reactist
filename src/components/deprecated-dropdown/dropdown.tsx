import React from 'react'
import ReactDOM from 'react-dom'
import classNames from 'classnames'

import Button from '../deprecated-button'

import './dropdown.less'

type BoxProps = {
    onShowBody?: () => void
    onHideBody?: () => void
    allowBodyInteractions?: boolean
    top?: boolean
    right?: boolean
    scrolling_parent?: string
    children?: [
        React.ReactElement<TriggerProps>,
        React.ReactElement<BodyProps> | ((props: BodyProps) => JSX.Element),
    ]
    className?: string
}

type BoxState = {
    top: boolean
    showBody: boolean
}

class Box extends React.Component<React.PropsWithChildren<BoxProps>, BoxState> {
    public static displayName: string

    constructor(props: BoxProps, context: React.Context<unknown>) {
        super(props, context)
        this.state = {
            showBody: false,
            top: props.top || false,
        }

        this._timeout = undefined
    }

    componentWillUnmount() {
        document.removeEventListener('click', this._handleClickOutside, true)
        if (this._timeout) {
            clearTimeout(this._timeout)
        }
    }
    _timeout?: ReturnType<typeof setTimeout>

    _handleClickOutside = (event: MouseEvent) => {
        const dropdownDOMNode = ReactDOM.findDOMNode(this)

        if (dropdownDOMNode && !dropdownDOMNode.contains(event.target as Node))
            this._toggleShowBody()
        else if (!this.props.allowBodyInteractions) {
            // won't close when body interactions are allowed
            this._timeout = setTimeout(() => {
                if (this.state.showBody) {
                    this._toggleShowBody()
                }
            }, 100)
        }
    }

    _toggleShowBody = () => {
        if (!this.state.showBody) {
            // will show
            if (this.props.onShowBody) this.props.onShowBody()
            document.addEventListener('click', this._handleClickOutside, true)
        } else {
            // will hide
            if (this.props.onHideBody) this.props.onHideBody()
            document.removeEventListener('click', this._handleClickOutside, true)
        }

        this.setState({
            showBody: !this.state.showBody,
        })
    }

    _getTriggerComponent() {
        const _trigger = this.props.children?.[0]
        return _trigger
            ? React.cloneElement(_trigger, { onClick: this._toggleShowBody })
            : undefined
    }

    // https://facebook.github.io/react/docs/refs-and-the-dom.html#exposing-dom-refs-to-parent-components
    _setPosition = (body: HTMLElement | null) => {
        if (body) {
            const scrollingParent = document.getElementById(
                this.props.scrolling_parent ? this.props.scrolling_parent : '',
            )

            if (scrollingParent) {
                const dropdown = ReactDOM.findDOMNode(this)
                if (!dropdown) {
                    return
                }
                const dropdownVerticalPosition = (ReactDOM.findDOMNode(this) as HTMLElement)
                    .offsetTop
                const dropdownTrigger = (dropdown as Element).querySelector('.trigger')
                if (!dropdownTrigger) {
                    return
                }
                const dropdownTriggerHeight = dropdownTrigger.clientHeight
                const dropdownBodyHeight = body.clientHeight

                const scrollingParentHeight = scrollingParent.clientHeight
                const scrollingParentOffset = scrollingParent.scrollTop

                const bottomOffset =
                    scrollingParentHeight +
                    scrollingParentOffset -
                    dropdownVerticalPosition -
                    dropdownTriggerHeight

                const top = bottomOffset < dropdownBodyHeight

                if (top !== this.state.top) {
                    this.setState({ top })
                }
            }
        }
    }

    _getBodyComponent() {
        if (!this.state.showBody) {
            return null
        }
        const { top } = this.state
        const { right = false, children } = this.props
        const props = { top, right, setPosition: this._setPosition }

        const className = classNames({
            body_wrapper: true,
            with_arrow: true,
            top: top,
            bottom: !top,
        })

        const body = children?.[1]

        const contentMarkup =
            typeof body === 'function'
                ? body(props)
                : body
                ? React.cloneElement(body, props)
                : undefined
        return (
            <div className={className} style={{ position: 'relative' }}>
                {contentMarkup}
            </div>
        )
    }

    render() {
        const className = classNames('reactist_dropdown', this.props.className)
        const { top } = this.state

        return (
            <div
                style={{ display: 'inline-block' }}
                className={className}
                data-testid="reactist-dropdown-box"
            >
                {top && this._getBodyComponent()}
                {this._getTriggerComponent()}
                {!top && this._getBodyComponent()}
            </div>
        )
    }
}

Box.displayName = 'Dropdown.Box'

type NativeButtonProps = React.DetailedHTMLProps<
    React.ButtonHTMLAttributes<HTMLButtonElement>,
    HTMLButtonElement
>

type TriggerProps = Omit<NativeButtonProps, 'title' | 'onClick'> & {
    tooltip?: React.ReactNode
    /**
     * @private the onClick prop is not to be used externally
     */
    onClick?: NativeButtonProps['onClick']
}

const Trigger = React.forwardRef<HTMLButtonElement, TriggerProps>(function Trigger(
    { children, onClick, tooltip, className, ...props },
    ref,
) {
    function handleClick(event: React.MouseEvent<HTMLButtonElement, MouseEvent>) {
        event.preventDefault()
        event.stopPropagation()
        if (onClick) onClick(event)
    }

    return (
        <Button
            {...props}
            className={classNames('trigger', className)}
            onClick={handleClick}
            tooltip={tooltip}
            ref={ref}
        >
            {children}
        </Button>
    )
})

Trigger.displayName = 'Dropdown.Trigger'

type BodyProps = {
    setPosition?: React.Ref<HTMLDivElement>
    children?: React.ReactNode
    top?: boolean
    right?: boolean
}

function Body({ top, right, children, setPosition }: BodyProps) {
    const style: React.CSSProperties = { position: 'absolute', right: 0, top: 0 }

    if (top) {
        style.top = 'auto'
        style.bottom = 0
    }

    if (right) {
        style.right = 'auto'
        style.left = 0
    }

    return (
        <div
            ref={setPosition}
            style={style}
            className="body"
            id="reactist-dropdown-body"
            data-testid="reactist-dropdown-body"
        >
            {children}
        </div>
    )
}

Body.displayName = 'Dropdown.Body'

const Dropdown = {
    Box,
    Trigger,
    Body,
}

export { Dropdown }
