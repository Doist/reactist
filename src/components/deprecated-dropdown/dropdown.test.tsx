import React from 'react'

import { Dropdown } from './dropdown'
import { render, screen } from '@testing-library/react'
import userEvent from '@testing-library/user-event'

describe('Dropdown', () => {
    describe('Dropdown.Box', () => {
        it('renders the Trigger component without crashing', () => {
            const { container } = render(
                <Dropdown.Box>
                    <Dropdown.Trigger />
                    <Dropdown.Body />
                </Dropdown.Box>,
            )

            expect(container).toMatchSnapshot()
        })

        it('applies additionally provided class name', () => {
            render(
                <Dropdown.Box className="additional class">
                    <Dropdown.Trigger />
                    <Dropdown.Body />
                </Dropdown.Box>,
            )

            const box = screen.getByTestId('reactist-dropdown-box')

            expect(box).toHaveClass('additional class')
        })

        it('sets the correct positioning style attributes to the Body component', () => {
            render(
                <Dropdown.Box top right>
                    <Dropdown.Trigger>Click me</Dropdown.Trigger>
                    <Dropdown.Body />
                </Dropdown.Box>,
            )

            userEvent.click(screen.getByRole('button', { name: 'Click me' }))

            const body = screen.getByTestId('reactist-dropdown-body')

            expect(body).toHaveStyle({ top: 'auto', right: 'auto', left: 0, bottom: 0 })
        })

        it('toggles the Body component with each Trigger click', () => {
            render(
                <Dropdown.Box>
                    <Dropdown.Trigger>Click me</Dropdown.Trigger>
                    <Dropdown.Body>My content</Dropdown.Body>
                </Dropdown.Box>,
            )

            expect(screen.queryByTestId('reactist-dropdown-body')).not.toBeInTheDocument()

            userEvent.click(screen.getByRole('button', { name: 'Click me' }))
            expect(screen.getByTestId('reactist-dropdown-body')).toBeVisible()
            expect(screen.getByText('My content')).toBeVisible()

            userEvent.click(screen.getByRole('button', { name: 'Click me' }))
            expect(screen.queryByTestId('reactist-dropdown-body')).not.toBeInTheDocument()
            expect(screen.queryByText('My content')).not.toBeInTheDocument()
        })

        it('toggles the Body component as a function with each Trigger click', () => {
            render(
                <Dropdown.Box>
                    <Dropdown.Trigger>Click me</Dropdown.Trigger>
                    {(props) => <Dropdown.Body {...props}>My content</Dropdown.Body>}
                </Dropdown.Box>,
            )

            expect(screen.queryByTestId('reactist-dropdown-body')).not.toBeInTheDocument()

            userEvent.click(screen.getByRole('button', { name: 'Click me' }))
            expect(screen.getByTestId('reactist-dropdown-body')).toBeVisible()

            userEvent.click(screen.getByRole('button', { name: 'Click me' }))
            expect(screen.queryByTestId('reactist-dropdown-body')).not.toBeInTheDocument()
        })

        it('renders the Trigger component first when top prop is not provided', () => {
            const { container } = render(
                <Dropdown.Box>
                    <Dropdown.Trigger>Click me</Dropdown.Trigger>
                    <Dropdown.Body />
                </Dropdown.Box>,
            )

            userEvent.click(screen.getByRole('button', { name: 'Click me' }))

            expect(container).toMatchSnapshot()
        })

        it('renders the Body component first when top prop is provided', () => {
            const { container } = render(
                <Dropdown.Box top>
                    <Dropdown.Trigger>Click me</Dropdown.Trigger>
                    <Dropdown.Body />
                </Dropdown.Box>,
            )

            userEvent.click(screen.getByRole('button', { name: 'Click me' }))

            expect(container).toMatchSnapshot()
        })

        it('calls onShowBody and onHideBody callbacks when the Body component is shown and hidden', () => {
            const onShowBodySpy = jest.fn()
            const onHideBodySpy = jest.fn()

            render(
                <Dropdown.Box onShowBody={onShowBodySpy} onHideBody={onHideBodySpy}>
                    <Dropdown.Trigger>Click me</Dropdown.Trigger>
                    <Dropdown.Body />
                </Dropdown.Box>,
            )

            userEvent.click(screen.getByRole('button', { name: 'Click me' }))
            expect(onShowBodySpy).toHaveBeenCalledTimes(1)
            expect(onHideBodySpy).not.toHaveBeenCalled()

            userEvent.click(screen.getByRole('button', { name: 'Click me' }))
            expect(onShowBodySpy).toHaveBeenCalledTimes(1)
            expect(onHideBodySpy).toHaveBeenCalledTimes(1)
        })
    })
})
