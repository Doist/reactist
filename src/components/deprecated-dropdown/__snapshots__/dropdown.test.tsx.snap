// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Dropdown Dropdown.Body renders all children without crashing 1`] = `
<div
  className="body"
  id="reactist-dropdown-body"
  style={
    Object {
      "position": "absolute",
      "right": 0,
      "top": 0,
    }
  }
>
  Body Content
</div>
`;

exports[`Dropdown Dropdown.Body renders all children without crashing 2`] = `
<div
  className="body"
  id="reactist-dropdown-body"
  style={
    Object {
      "position": "absolute",
      "right": 0,
      "top": 0,
    }
  }
>
  <div>
    <img
      alt=""
      src=""
    />
    Some additional text as well
  </div>
</div>
`;

exports[`Dropdown Dropdown.Body renders empty without crashing 1`] = `
<div
  className="body"
  id="reactist-dropdown-body"
  style={
    Object {
      "position": "absolute",
      "right": 0,
      "top": 0,
    }
  }
/>
`;

exports[`Dropdown Dropdown.Box renders the Body component first when top prop is provided 1`] = `
<Dropdown.Box
  top={true}
>
  <div
    className="reactist_dropdown"
    style={
      Object {
        "display": "inline-block",
      }
    }
  >
    <div
      className="body_wrapper with_arrow top"
      style={
        Object {
          "position": "relative",
        }
      }
    >
      <Dropdown.Body
        right={false}
        setPosition={[Function]}
        top={true}
      >
        <div
          className="body"
          id="reactist-dropdown-body"
          style={
            Object {
              "bottom": 0,
              "position": "absolute",
              "right": 0,
              "top": "auto",
            }
          }
        />
      </Dropdown.Body>
    </div>
    <Dropdown.Trigger
      onClick={[Function]}
    >
      <Button
        className="trigger"
        disabled={false}
        loading={false}
        onClick={[Function]}
        size="default"
      >
        <button
          aria-disabled={false}
          className="reactist_button trigger"
          onClick={[Function]}
          type="button"
        />
      </Button>
    </Dropdown.Trigger>
  </div>
</Dropdown.Box>
`;

exports[`Dropdown Dropdown.Box renders the Trigger component first when top prop is not provided 1`] = `
<Dropdown.Box>
  <div
    className="reactist_dropdown"
    style={
      Object {
        "display": "inline-block",
      }
    }
  >
    <Dropdown.Trigger
      onClick={[Function]}
    >
      <Button
        className="trigger"
        disabled={false}
        loading={false}
        onClick={[Function]}
        size="default"
      >
        <button
          aria-disabled={false}
          className="reactist_button trigger"
          onClick={[Function]}
          type="button"
        />
      </Button>
    </Dropdown.Trigger>
    <div
      className="body_wrapper with_arrow bottom"
      style={
        Object {
          "position": "relative",
        }
      }
    >
      <Dropdown.Body
        right={false}
        setPosition={[Function]}
        top={false}
      >
        <div
          className="body"
          id="reactist-dropdown-body"
          style={
            Object {
              "position": "absolute",
              "right": 0,
              "top": 0,
            }
          }
        />
      </Dropdown.Body>
    </div>
  </div>
</Dropdown.Box>
`;

exports[`Dropdown Dropdown.Box renders the Trigger component without crashing 1`] = `
<Dropdown.Box>
  <div
    className="reactist_dropdown"
    style={
      Object {
        "display": "inline-block",
      }
    }
  >
    <Dropdown.Trigger
      onClick={[Function]}
    >
      <Button
        className="trigger"
        disabled={false}
        loading={false}
        onClick={[Function]}
        size="default"
      >
        <button
          aria-disabled={false}
          className="reactist_button trigger"
          onClick={[Function]}
          type="button"
        />
      </Button>
    </Dropdown.Trigger>
  </div>
</Dropdown.Box>
`;

exports[`Dropdown Dropdown.Trigger renders all children without crashing 1`] = `
<Button
  className="trigger"
  disabled={false}
  loading={false}
  onClick={[Function]}
  size="default"
>
  Trigger Content
</Button>
`;

exports[`Dropdown Dropdown.Trigger renders all children without crashing 2`] = `
<Button
  className="trigger"
  disabled={false}
  loading={false}
  onClick={[Function]}
  size="default"
>
  <div>
    <img
      alt=""
      src=""
    />
    Some additional text as well
  </div>
</Button>
`;

exports[`Dropdown Dropdown.Trigger renders empty without crashing 1`] = `
<Button
  className="trigger"
  disabled={false}
  loading={false}
  onClick={[Function]}
  size="default"
/>
`;
